scoring/score.py

```diff
@@ -45,6 +45,49 @@ def calculate_scores(self):
         }
 
 
+class EndStateScorer(object):
+    # Assumption: we will trust the *ordering* of the entries in the data we
+    # are given and *ignore* the actual 'time' values. We defer handling of
+    # equialent-time operations to the territory controller.
+
+    def __init__(self, teams_data, arena_data):
+        self._zone_to_tla = {
+            info['zone']: tla
+            for tla, info in teams_data.items()
+        }
+
+        # Mapping from station_code -> owning zone
+        self._final_state = {
+            claim['station_code']: claim['zone']
+            for claim in arena_data['other']['territory_claims']
+        }
```

Comment:
This should probably be in `calculate_scores` since it relates to the core logic of the score calculation.

Reply:
https://github.com/srobo/sr2021-comp/pull/4/commits/9238bb165c8aa844edfe137ab0766fe706b76ab6